intra_process_demo/CMakeLists.txt

```diff
@@ -94,3 +92,60 @@ macro(targets)
 endmacro()
 
 call_for_each_rmw_implementation(targets GENERATE_DEFAULT)
+
+if(AMENT_ENABLE_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+
+  if(NOT WIN32)
+    set(test_executable_extension "")
+    set(test_executable_subfolder "")
+  else()
+    set(test_executable_extension ".exe")
+    set(test_executable_subfolder "/${CMAKE_BUILD_TYPE}")
+    if("${test_executable_subfolder}" STREQUAL "/")
+      set(test_executable_subfolder "/Debug")
+    endif()
+  endif()
+
+  # Add each test case.  Multi-executable tests can be specified in
+  # semicolon-separated strings, like  exe1:exe2.
+  set(demo_tests
+    cyclic_pipeline
+    two_node_pipeline)
+
+  macro(tests)
+    foreach(demo_test ${demo_tests})
+      string(REPLACE ":" ";" demo_executables "${demo_test}")
+      set(RCLCPP_DEMOS_EXPECTED_OUTPUT "")
+      foreach(executable ${demo_executables})
+        list(APPEND RCLCPP_DEMOS_EXPECTED_OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/test/${executable}")
+      endforeach()
+
+      # get typesupport of rmw implementation to include / link against the corresponding interfaces
+      get_rmw_typesupport(typesupport_impl "${rmw_implementation}")
+
+      set(RCLCPP_DEMOS_EXECUTABLE "")
+      foreach(executable ${demo_executables})
+        list(APPEND RCLCPP_DEMOS_EXECUTABLE "${CMAKE_CURRENT_BINARY_DIR}${test_executable_subfolder}/${executable}${target_suffix}${test_executable_extension}")
```

Comment:
Yeah, I originally used the approach you describe in the tests or the examples (I don't remember well right now), but ended up not using a generator expression because of the reasons you mention.